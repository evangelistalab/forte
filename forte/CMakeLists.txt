if (POLICY CMP0025)
  cmake_policy(SET CMP0025 NEW)
endif ()

cmake_policy(SET CMP0048 NEW)  # project_VERSION* variables populated from project(... VERSION x.x.x) string

project(forte
        VERSION 0.1
        LANGUAGES CXX C)
set(forte_AUTHORS      "Francesco A. Evangelista and Group")
set(forte_DESCRIPTION  "Quantum Chemistry Methods for Strongly Correlated Electrons plugin to Psi4")
set(forte_URL          "https://github.com/evangelistalab/forte")
set(forte_LICENSE      "GPL-3.0+")
set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
set(TargetOpenMP_FIND_COMPONENTS "CXX")

# cmake_minimum_required(VERSION 3.15 FATAL_ERROR)
list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)

include(psi4OptionsTools)

option_with_print(ENABLE_CheMPS2 "Enable CheMPS2 solver" OFF)
option_with_print(ENABLE_OPENMP "Enable threadsafe linking to OpenMP parallelized programs" ON)
option_with_print(ENABLE_MPI "Enable MPI parallelization" OFF)
option_with_print(ENABLE_GA "Enable Global Arrays" OFF)
option_with_print(MAX_DET_ORB "Set the maximum number of orbitals in a determinant" OFF)
option_with_print(ENABLE_CODECOV "Enable compilation with code coverage flags" OFF)
option_with_print(ENABLE_UNTESTED_CODE "Enable code not covered by code coverage" OFF)

include(autocmake_omp)  # no longer useful, probably need to copy psi4/external/common/lapack to cmake
include(autocmake_mpi)  # MPI option A

find_package(psi4 1.3 REQUIRED)
find_package(pybind11 CONFIG REQUIRED)
find_package(Python COMPONENTS Interpreter Development REQUIRED)
find_package(TargetLAPACK REQUIRED)
find_package(ambit 0.5 REQUIRED)

if(ENABLE_CheMPS2)
    find_package(CheMPS2 1.8.3 CONFIG REQUIRED)
endif()

set(Boost_USE_STATIC_LIBS OFF)
set(Boost_USE_MULTITHREADED ON)
set(Boost_USE_STATIC_RUNTIME OFF)
find_package(Boost REQUIRED)

if(Boost_FOUND)
    include_directories(${Boost_INCLUDE_DIRS})
endif()

# Including custom cmake rules
include(cmake/clang-cxx-dev-tools.cmake)
include(cmake/git-version.cmake)

# Check if we can compile with SSE4.2
check_cxx_compiler_flag("-msse4.2" SSE42_FLAG)
if(SSE42_FLAG)
    message("-- Adding SSE4.2 Flag")
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2")
endif()
add_compile_options(-Wall -Wextra -pedantic) # -Werror)

if(ENABLE_CODECOV)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage")
endif()

if(ENABLE_UNTESTED_CODE)
    add_definitions(-DENABLE_UNTESTED_CODE)
endif()

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c++1z-extensions") # avoid warnings for C++17

# List of CC files
pybind11_add_module(forte
api/ambit_api.cc
api/cube_file_api.cc
api/integrals_api.cc
api/mospaceinfo_api.cc
api/rdms_api.cc
api/options_api.cc
api/orbital_api.cc
api/sci_api.cc
api/state_info_api.cc
api/forte_python_module.cc
base_classes/active_space_method.cc
base_classes/active_space_solver.cc
base_classes/dynamic_correlation_solver.cc
base_classes/forte_options.cc
base_classes/mo_space_info.cc
base_classes/orbital_transform.cc
base_classes/rdms.cc
base_classes/scf_info.cc
base_classes/state_info.cc
casscf/casscf.cc
casscf/casscf_gradient.cc
casscf/casscf_orb_grad.cc
casscf/casscf_orb_grad_deriv.cc
casscf/cpscf.cc
casscf/mcscf_2step.cc
ci_ex_states/excited_state_solver.cc
ci_rdm/ci_rdms.cc
ci_rdm/ci_rdms_dynamic.cc
dmrg/dmrgscf.cc
dmrg/dmrgsolver.cc
fci/binary_graph.cc
fci/fci_solver.cc
fci/fci_vector.cc
fci/fci_vector_h_diag.cc
fci/fci_vector_hamiltonian.cc
fci/fci_vector_rdm.cc
fci/string_hole_list.cc
fci/string_lists.cc
fci/string_oo_list.cc
fci/string_vo_list.cc
fci/string_vvoo_list.cc
finite_temperature/finite_temperature.cc
forte.cc
gradient_tpdm/backtransform_tpdm.cc
gradient_tpdm/integraltransform_tpdm_unrestricted.cc
gradient_tpdm/integraltransform_tpdm_restricted.cc
helpers/blockedtensorfactory.cc
helpers/combinatorial.cc
helpers/cube_file.cc
helpers/disk_io.cc
helpers/helpers.cc
helpers/symmetry.cc
helpers/iterative_solvers.cc
helpers/lbfgs/lbfgs.cc
helpers/lbfgs/lbfgs_param.cc
helpers/lbfgs/rosenbrock.cc
helpers/printing.cc
helpers/string_algorithms.cc
integrals/active_space_integrals.cc
integrals/cholesky_integrals.cc
integrals/conventional_integrals.cc
integrals/custom_integrals.cc
integrals/df_integrals.cc
integrals/diskdf_integrals.cc
integrals/distribute_df_integrals.cc
integrals/integrals.cc
integrals/integrals_psi4_interface.cc
integrals/make_integrals.cc
integrals/parallel_ccvv_algorithms.cc
integrals/paralleldfmo.cc
mrdsrg-helper/dsrg_mem.cc
mrdsrg-helper/dsrg_source.cc
mrdsrg-helper/dsrg_time.cc
mrdsrg-helper/dsrg_transformed.cc
mrdsrg-helper/run_dsrg.cc
mrdsrg-so/mrdsrg_so.cc
mrdsrg-so/so-mrdsrg.cc
mrdsrg-spin-adapted/dsrg_mrpt.cc
mrdsrg-spin-adapted/dsrg_mrpt_2nd.cc
mrdsrg-spin-adapted/dsrg_mrpt_amp.cc
mrdsrg-spin-adapted/dsrg_mrpt_comm.cc
mrdsrg-spin-adapted/sadsrg.cc
mrdsrg-spin-adapted/sadsrg_amps_analysis.cc
mrdsrg-spin-adapted/sadsrg_block_labels.cc
mrdsrg-spin-adapted/sadsrg_comm.cc
mrdsrg-spin-adapted/sa_dsrgpt.cc
mrdsrg-spin-adapted/sa_ldsrg2.cc
mrdsrg-spin-adapted/sa_mrdsrg.cc
mrdsrg-spin-adapted/sa_mrdsrg_amps.cc
mrdsrg-spin-adapted/sa_mrdsrg_diis.cc
mrdsrg-spin-adapted/sa_mrpt2.cc
mrdsrg-spin-adapted/sa_mrpt3.cc
mrdsrg-spin-integrated/active_dsrgpt2.cc
mrdsrg-spin-integrated/dsrg_mrpt2.cc
mrdsrg-spin-integrated/dsrg_mrpt2_ms.cc
mrdsrg-spin-integrated/dsrg_mrpt3.cc
mrdsrg-spin-integrated/dwms_mrpt2.cc
mrdsrg-spin-integrated/master_mrdsrg.cc
mrdsrg-spin-integrated/mcsrgpt2_mo.cc
mrdsrg-spin-integrated/mrdsrg.cc
mrdsrg-spin-integrated/mrdsrg_amplitude.cc
mrdsrg-spin-integrated/mrdsrg_commutator.cc
mrdsrg-spin-integrated/mrdsrg_diis.cc
mrdsrg-spin-integrated/mrdsrg_nonpt.cc
mrdsrg-spin-integrated/mrdsrg_pt.cc
mrdsrg-spin-integrated/mrdsrg_smart_s.cc
mrdsrg-spin-integrated/mrdsrg_srg.cc
mrdsrg-spin-integrated/three_dsrg_mrpt2.cc
orbital-helpers/ao_helper.cc
orbital-helpers/aosubspace.cc
orbital-helpers/ci-no/ci-no.cc
orbital-helpers/ci-no/mrci-no.cc
orbital-helpers/fragment_projector.cc
orbital-helpers/iao_builder.cc
orbital-helpers/localize.cc
orbital-helpers/mp2_nos.cc
orbital-helpers/orbital_embedding.cc
orbital-helpers/orbitaloptimizer.cc
orbital-helpers/pao_builder.cc
orbital-helpers/semi_canonicalize.cc
orbital-helpers/unpaired_density.cc
pci/pci.cc
pci/pci_sigma.cc
post_process/spin_corr.cc
sci/aci.cc
sci/aci_build_F.cc
sci/gasaci_build_F.cc
sci/asci.cc
sci/detci.cc
sci/fci_mo.cc
sci/mrpt2.cc
sci/sci.cc
sparse_ci/ci_reference.cc
sparse_ci/determinant_functions.cc
sparse_ci/determinant_hashvector.cc
sparse_ci/determinant_substitution_lists.cc
sparse_ci/sigma_vector.cc
sparse_ci/sigma_vector_dynamic.cc
sparse_ci/sigma_vector_sparse_list.cc
sparse_ci/sorted_string_list.cc
sparse_ci/sparse_ci_solver.cc
sparse_ci/sparse_operator.cc
sparse_ci/sparse_exp.cc
sparse_ci/sparse_fact_exp.cc
sparse_ci/sparse_hamiltonian.cc
sparse_ci/sq_operator.cc
sparse_ci/sparse_state_vector.cc
v2rdm/v2rdm.cc
)

target_link_libraries(forte PRIVATE psi4::core)
target_link_libraries(forte PRIVATE tgt::MathOpenMP)
#set_target_properties(forte PROPERTIES PREFIX "")
target_link_libraries(forte PRIVATE ambit::ambit)
if(TARGET CheMPS2::chemps2)
    target_link_libraries(forte PRIVATE CheMPS2::chemps2)
    add_definitions(-DHAVE_CHEMPS2)
endif()

target_include_directories(forte PRIVATE .)

# 64bit implementation
if(MAX_DET_ORB)
    add_definitions(-DMAX_DET_ORB=${MAX_DET_ORB})
else()
    add_definitions(-DMAX_DET_ORB=64)
endif()

if(ENABLE_MPI)
    target_link_libraries(forte PRIVATE ${MPI_CXX_LIBRARIES})  # MPI option A
    #target_link_libraries(forte PRIVATE MPI::MPI_CXX)  # MPI option B
    #target_add_definitions(forte PRIVATE HAVE_MPI)  # MPI option B
endif()

if(ENABLE_GA)
    target_link_libraries(forte PRIVATE GlobalArrays::ga)
endif()
