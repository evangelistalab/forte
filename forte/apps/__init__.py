from .mcscf import run_mcscf
from .hf import run_hf
