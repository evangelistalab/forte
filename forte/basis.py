class Basis:
    """
    A class used to store basis information.
    """
    def __init__(self, basis):
        """
        initialize a Basis object

        Parameters
        ----------
        basis : str
            a basis object
        """
        self._basis = basis

    def __repr__(self):
        """
        return a string representation of this object
        """
        return f'Basis(\'{self._basis}\')'

    def __str__(self):
        """
        return a string representation of this object
        """
        return self._basis
