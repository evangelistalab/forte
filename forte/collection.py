class Collection:
    """
    A class to that handles forte objects

    Attributes
    ----------
    """
    def __init__(self, scf_info=None, mospaceinfo=None, ints=None, psi_wfn=None):
        """
        initialize a Basis object

        Parameters
        ----------
        restricted : bool
            a basis object
        """
        self._collection = {}
        self._collection['scf_info'] = scf_info
        self._collection['mospaceinfo'] = mospaceinfo
        self._collection['ints'] = ints
        self._collection['psi_wfn'] = psi_wfn

    def __repr__(self):
        """
        return a string representation of this object
        """
        return f'Collection()'

    def __str__(self):
        """
        return a string representation of this object
        """
        return f'Collection()'

    @property
    def scf_info(self):
        return self._collection['scf_info']

    @scf_info.setter
    def scf_info(self, val):
        self._collection['scf_info'] = val

    @property
    def mospaceinfo(self):
        return self._collection['mospaceinfo']

    @property
    def ints(self):
        return self._collection['ints']

    @property
    def psi_wfn(self):
        return self._collection['psi_wfn']

    @psi_wfn.setter
    def psi_wfn(self, val):
        self._collection['psi_wfn'] = val
