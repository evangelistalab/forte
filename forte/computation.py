# from .results import Results

# class Computation:
#     """
#     A class used to represent a computation.

#     Attributes
#     ----------
#     basis : str
#         the basis name
#     """
#     def __init__(self, model=None, method=None):
#         """
#         initialize a Basis object

#         Parameters
#         ----------
#         basis : str
#             a basis object
#         """
#         self._model = model
#         self._method = method

#     def __repr__(self):
#         """
#         return a string representation of this object
#         """
#         return f'Computation(model={repr(self._model)},method={repr(self._method)})'

#     def __str__(self):
#         """
#         return a string representation of this object
#         """
#         return self._basis

#     @property
#     def model(self):
#         return self._model

#     @property
#     def method(self):
#         return self._method

#     def energy(self):
#         """Compute the energy."""
#         results = Results()
#         self._method.energy(self._model, results=results)
#         return results
