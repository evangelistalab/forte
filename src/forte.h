/*
 * @BEGIN LICENSE
 *
 * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
 * that implements a variety of quantum chemistry methods for strongly
 * correlated electrons.
 *
 * Copyright (c) 2012-2019 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
 *
 * The copyrights for code used from other parties are included in
 * the corresponding files.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see http://www.gnu.org/licenses/.
 *
 * @END LICENSE
 */

#include "base_classes/forte_options.h"

namespace forte {

class ForteIntegrals;

void forte_options(ForteOptions& options);

std::pair<int, int> startup();
void banner();
void cleanup();

void read_options(ForteOptions& options);
psi::SharedWavefunction run_forte(psi::SharedWavefunction ref_wfn, psi::Options& options);

std::shared_ptr<MOSpaceInfo> make_mo_space_info(psi::SharedWavefunction ref_wfn,
                                                std::shared_ptr<ForteOptions> options);

psi::SharedMatrix make_aosubspace_projector(psi::SharedWavefunction ref_wfn, psi::Options& options);

void make_ci_nos(psi::SharedWavefunction ref_wfn, psi::Options& options,
                 std::shared_ptr<ForteIntegrals> ints, std::shared_ptr<MOSpaceInfo> mo_space_info);

double forte_old_methods(psi::SharedWavefunction ref_wfn, psi::Options& options,
                         std::shared_ptr<ForteIntegrals> ints,
                         std::shared_ptr<MOSpaceInfo> mo_space_info);
} // namespace forte
