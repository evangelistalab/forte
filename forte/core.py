#
# @BEGIN LICENSE
#
# Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
# that implements a variety of quantum chemistry methods for strongly
# correlated electrons.
#
# Copyright (c) 2012-2021 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
#
# The copyrights for code used from other parties are included in
# the corresponding files.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
# @END LICENSE
#

import logging
import psi4
import forte

logging_depth = 0


def clean_options():
    """
    A function to clear the options object

    This function does the following
    1. clears the psi4 and Forte options object
    2. re-registers all of Forte options (in their default values)
    3. pushes the value of Forte options to the psi4 options object
    """
    # clear options and allocate a fresh forte.forte_options object
    psi4.core.clean_options()
    forte.forte_options = forte.ForteOptions()

    # register options defined in Forte in the forte_options object
    forte.register_forte_options(forte.forte_options)

    # Push the options defined in forte_options to psi
    psi_options = psi4.core.get_options()
    psi_options.set_current_module('FORTE')
    forte.forte_options.push_options_to_psi4(psi_options)


class ForteManager(object):
    """Singleton class to handle startup and cleanup of forte (mostly ambit)"""
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = True
            # Put any initialization here.
            my_proc, n_nodes = forte.startup()
        return cls._instance

    def __del__(cls):
        forte.cleanup()


def start_logging(filename='forte.log', level=logging.DEBUG):
    """
    This function sets the output of logs to ``filename`` (default = forte.log)
    and sets the log level to all information.

    Parameters
    ----------
    filename: str
        the name of the log file (default = 'forte.log')
    level: {logging.DEBUG, logging.INFO, logging.WARNING, logging.ERROR, logging.CRITICAL}
        the level of severity of the events tracked (default = logging.DEBUG, which means track everything)
    """
    logging.basicConfig(filename=filename, level=level, format='# %(asctime)s | %(levelname)s | %(message)s')
    logging.info('Starting the Forte logger')


def flog(level, msg):
    """
    Log the message ``msg`` with logging level ``level``.

    ``level`` should be chosen in this way:
    debug: for detailed ouput mostly for diagnostic purpose
    info: for information produced during normal operation of the program
    warning: for a warning regarding a particular runtime event
    error: for an error that does not raise an exception

    Parameters
    ----------
    level: str
        the level of the message logged. Can be any of ('debug','info','warning','error')
    msg: str
        the text to be logged

    This function calls the logging module and puts some spaces before
    the message that we want to log.
    """
    global logging_depth
    level = level.lower()
    spaces = ' ' * max((logging_depth - 1), 0) * 2
    s = f"{spaces}{msg}"
    if level == 'info':
        logging.info(s)
    elif level == 'warning':
        logging.warning(s)
    elif level == 'debug':
        logging.debug(s)
    elif level == 'error':
        logging.error(s)
    else:
        raise ValueError(f'forte.core.flog was called with an unrecognized level ({level})')


def increase_log_depth(func):
    """
    This is a decorator used to increase the depth of forte's log.

    It is used to decorate the run() function of solvers:
        @increase_log_dept
            def run(self):
                ...
    """
    def wrapper(*args, **kwargs):
        global logging_depth
        logging_depth += 1
        func(*args, **kwargs)
        logging_depth += -1

    return wrapper
