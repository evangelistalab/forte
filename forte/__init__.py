#
# @BEGIN LICENSE
#
# Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
# that implements a variety of quantum chemistry methods for strongly
# correlated electrons.
#
# Copyright (c) 2012-2017 by its authors (see COPYING, COPYING.LESSER, AUTHORS).
#
# The copyrights for code used from other parties are included in
# the corresponding files.
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Lesser General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program.  If not, see http://www.gnu.org/licenses/.
#
# @END LICENSE
#
"""Plugin docstring.

"""

import sys
import logging

logging.basicConfig(filename='forte.log', level=logging.INFO)
logging.info('Starting the Forte logger')

# Load Python modules
from .molecule import Molecule
from .basis import Basis
from .results import Results
from .model import Model, MolecularModel

from .pymodule import *
from .register_forte_options import *
from .core import ForteManager, clean_options
from .forte import *

__version__ = '0.2.0'
__author__ = 'Forte Developers'

# Create a ForteOptions object (stores all options)
forte_options = forte.ForteOptions()

# Register options defined in Forte in the forte_options object
register_forte_options(forte_options)

# create a singleton to handle startup and cleanup of forte
ForteManager()

# If we are running psi4, push the options defined in forte_options to psi
if 'psi4' in sys.modules:
    psi_options = psi4.core.get_options()
    psi_options.set_current_module('FORTE')
    forte_options.push_options_to_psi4(psi_options)
