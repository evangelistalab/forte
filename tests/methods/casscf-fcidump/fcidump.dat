#! CASSCF on HF with no symmetry

import forte

refscf          = -99.87285247289

memory 500 mb

molecule HF{
  0 1
  F
  H  1 R
  R = 1.50
  symmetry c1
}

set globals {
    basis           cc-pvdz
    scf_type        out_of_core
    reference       RHF
    e_convergence   1e-10
    d_convergence   1e-10
    frozen_docc     [0]
    restricted_docc [4]
    active          [2]
}
scf,wfn = energy('scf', return_wfn=True)
fcidump(wfn,oe_ints=['EIGENVALUES'],write_pntgrp=True)
