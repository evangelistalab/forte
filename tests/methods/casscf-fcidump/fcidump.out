
    -----------------------------------------------------------------------
          Psi4: An Open-Source Ab Initio Electronic Structure Package
                               Psi4 undefined 

                         Git: Rev {fcidump_sym} 63eae05 dirty


    R. M. Parrish, L. A. Burns, D. G. A. Smith, A. C. Simmonett,
    A. E. DePrince III, E. G. Hohenstein, U. Bozkaya, A. Yu. Sokolov,
    R. Di Remigio, R. M. Richard, J. F. Gonthier, A. M. James,
    H. R. McAlexander, A. Kumar, M. Saitow, X. Wang, B. P. Pritchard,
    P. Verma, H. F. Schaefer III, K. Patkowski, R. A. King, E. F. Valeev,
    F. A. Evangelista, J. M. Turney, T. D. Crawford, and C. D. Sherrill,
    J. Chem. Theory Comput. 13(7) pp 3185--3197 (2017).
    (doi: 10.1021/acs.jctc.7b00174)


                         Additional Contributions by
    P. Kraus, H. Kruse, M. H. Lechner, M. C. Schieber, R. A. Shaw,
    A. Alenaizan, R. Galvelis, Z. L. Glick, S. Lehtola, and J. P. Misiewicz

    -----------------------------------------------------------------------


    Psi4 started on: Wednesday, 08 July 2020 09:46PM

    Process ID: 31164
    Host:       CHE-FEVANGE-01I (2)
    PSIDATADIR: /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4
    Memory:     500.0 MiB
    Threads:    1
    
  ==> Input File <==

--------------------------------------------------------------------------
#! CASSCF on HF with no symmetry

import forte

refscf          = -99.87285247289

memory 500 mb

molecule HF{
  0 1
  F
  H  1 R
  R = 1.50
  symmetry c1
}

set globals {
    basis           cc-pvdz
    scf_type        out_of_core
    reference       RHF
    e_convergence   1e-10
    d_convergence   1e-10
    frozen_docc     [0]
    restricted_docc [4]
    active          [2]
}
scf,wfn = energy('scf', return_wfn=True)
fcidump(wfn,oe_ints=['EIGENVALUES'],write_pntgrp=True)
--------------------------------------------------------------------------

  Memory set to 476.837 MiB by Python driver.

Scratch directory: /tmp/

*** tstart() called on CHE-FEVANGE-01I (2)
*** at Wed Jul  8 21:46:00 2020

   => Loading Basis Set <=

    Name: CC-PVDZ
    Role: ORBITAL
    Keyword: BASIS
    atoms 1 entry F          line   228 file /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4/basis/cc-pvdz.gbs 
    atoms 2 entry H          line    22 file /Users/fevange/Source/psi4/objdir-Debug/stage/share/psi4/basis/cc-pvdz.gbs 


         ---------------------------------------------------------
                                   SCF
               by Justin Turney, Rob Parrish, Andy Simmonett
                          and Daniel G. A. Smith
                              RHF Reference
                        1 Threads,    476 MiB Core
         ---------------------------------------------------------

  ==> Geometry <==

    Molecular point group: c1
    Full point group: C_inf_v

    Geometry (in Angstrom), charge = 0, multiplicity = 1:

       Center              X                  Y                   Z               Mass       
    ------------   -----------------  -----------------  -----------------  -----------------
         F            0.000000000000     0.000000000000    -0.075563346255    18.998403162730
         H            0.000000000000     0.000000000000     1.424436653745     1.007825032230

  Running in c1 symmetry.

  Rotational constants: A = ************  B =      7.82847  C =      7.82847 [cm^-1]
  Rotational constants: A = ************  B = 234691.66104  C = 234691.66104 [MHz]
  Nuclear repulsion =    3.175063264020000

  Charge       = 0
  Multiplicity = 1
  Electrons    = 10
  Nalpha       = 5
  Nbeta        = 5

  ==> Algorithm <==

  SCF Algorithm Type is OUT_OF_CORE.
  DIIS enabled.
  MOM disabled.
  Fractional occupation disabled.
  Guess Type is SAD.
  Energy threshold   = 1.00e-10
  Density threshold  = 1.00e-10
  Integral threshold = 0.00e+00

  ==> Primary Basis <==

  Basis Set: CC-PVDZ
    Blend: CC-PVDZ
    Number of shells: 9
    Number of basis function: 19
    Number of Cartesian functions: 20
    Spherical Harmonics?: true
    Max angular momentum: 2

  ==> Integral Setup <==

  ==> DiskJK: Disk-Based J/K Matrices <==

    J tasked:                  Yes
    K tasked:                  Yes
    wK tasked:                  No
    Memory [MiB]:              357
    Schwarz Cutoff:          1E-12

  Minimum eigenvalue in the overlap matrix is 1.3340269817E-01.
  Reciprocal condition number of the overlap matrix is 5.7946734818E-02.
    Using symmetric orthogonalization.

  ==> Pre-Iterations <==

  SCF Guess: Superposition of Atomic Densities via on-the-fly atomic UHF (no occupation information).

   -------------------------
    Irrep   Nso     Nmo    
   -------------------------
     A         19      19 
   -------------------------
    Total      19      19
   -------------------------

  ==> Iterations <==

                        Total Energy        Delta E     RMS |[F,P]|

   @RHF iter SAD:   -99.27377642284318   -9.92738e+01   0.00000e+00 
   @RHF iter   1:   -99.77292217081036   -4.99146e-01   2.17770e-02 DIIS
   @RHF iter   2:   -99.82685211910642   -5.39299e-02   2.06500e-02 DIIS
   @RHF iter   3:   -99.87047408765770   -4.36220e-02   2.92061e-03 DIIS
   @RHF iter   4:   -99.87267422470393   -2.20014e-03   8.77453e-04 DIIS
   @RHF iter   5:   -99.87284986388013   -1.75639e-04   8.42571e-05 DIIS
   @RHF iter   6:   -99.87285244146348   -2.57758e-06   1.08228e-05 DIIS
   @RHF iter   7:   -99.87285247416509   -3.27016e-08   1.46985e-06 DIIS
   @RHF iter   8:   -99.87285247448804   -3.22956e-10   1.88678e-07 DIIS
   @RHF iter   9:   -99.87285247449552   -7.47491e-12   1.53054e-08 DIIS
   @RHF iter  10:   -99.87285247449555   -2.84217e-14   4.68088e-10 DIIS
   @RHF iter  11:   -99.87285247449566   -1.13687e-13   6.57326e-11 DIIS
  Energy and wave function converged.


  ==> Post-Iterations <==

    Orbital Energies [Eh]
    ---------------------

    Doubly Occupied:                                                      

       1A    -26.279422     2A     -1.484580     3A     -0.593977  
       4A     -0.593977     5A     -0.565055  

    Virtual:                                                              

       6A      0.016621     7A      0.576770     8A      1.315282  
       9A      1.315282    10A      1.463904    11A      1.599282  
      12A      1.599282    13A      1.633070    14A      2.300773  
      15A      4.038213    16A      4.038213    17A      4.041123  
      18A      4.041123    19A      4.220284  

    Final Occupation by Irrep:
              A 
    DOCC [     5 ]

  @RHF Final Energy:   -99.87285247449566

   => Energetics <=

    Nuclear Repulsion Energy =              3.1750632640200003
    One-Electron Energy =                -147.0618097618362583
    Two-Electron Energy =                  44.0138940233205957
    Total Energy =                        -99.8728524744956587

Computation Completed


Properties will be evaluated at   0.000000,   0.000000,   0.000000 [a0]

Properties computed using the SCF density matrix

  Nuclear Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:     1.4066

  Electronic Dipole Moment: [e a0]
     X:    -0.0000      Y:     0.0000      Z:    -0.2428

  Dipole Moment: [e a0]
     X:    -0.0000      Y:     0.0000      Z:     1.1639     Total:     1.1639

  Dipole Moment: [D]
     X:    -0.0000      Y:     0.0000      Z:     2.9583     Total:     2.9583


*** tstop() called on CHE-FEVANGE-01I (2) at Wed Jul  8 21:46:00 2020
Module time:
	user time   =       0.49 seconds =       0.01 minutes
	system time =       0.03 seconds =       0.00 minutes
	total time  =          0 seconds =       0.00 minutes
Total time:
	user time   =       0.49 seconds =       0.01 minutes
	system time =       0.03 seconds =       0.00 minutes
	total time  =          0 seconds =       0.00 minutes
Writing integrals in FCIDUMP format to INTDUMP
	Presorting SO-basis two-electron integrals.
	Sorting File: SO Ints (nn|nn) nbuckets = 1
	Transforming the one-electron integrals and constructing Fock matrices
	Starting first half-transformation.
	Sorting half-transformed integrals.
	First half integral transformation complete.
	Starting second half-transformation.
	Two-electron integral transformation complete.
Integral transformation complete!
Writing TEI integrals in FCIDUMP format to INTDUMP
Writing frozen core operator in FCIDUMP format to INTDUMP
Writing orbital energies in FCIDUMP format to INTDUMP
Writing frozen core + nuclear repulsion energy in FCIDUMP format to INTDUMP
Done generating INTDUMP with integrals in FCIDUMP format.

    Psi4 stopped on: Wednesday, 08 July 2020 09:46PM
    Psi4 wall time for execution: 0:00:00.70

*** Psi4 exiting successfully. Buy a developer a beer!
